app/main.py

```diff
@@ -1,2 +1,146 @@
+            self.__setitem__(key, value)
+
+    def __str__(self) -> str:
+        result_str = "{"
+
+        for i in range(len(self.hash_table)):
+
+            if self.hash_table[i] is not None:
+                result_str += (f"'{self.hash_table[i].key}': "
+                               f"{self.hash_table[i].value}, ")
+
+        return f"{result_str}"[:-2] + "}"
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        node = Node(key, value)
+        index = node.hash_key % self.capacity
+
+        if (index == len(self.hash_table)
+                and self.hash_table[index] is not None
+                and self.hash_table[index].key != key):
+            index = 0
+
+        while (self.hash_table[index] is not None
+               and self.hash_table[index].key != key):
+            index = (index + 1) % self.capacity
+
+        if self.hash_table[index] is None:
+            self.length += 1
+
+        self.hash_table[index] = node
+
+        if self.length / self.capacity >= LOAD_FACTOR:
+            self._resize()
+
+    def __getitem__(self, key: Hashable) -> Any:
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+
+        while (self.hash_table[index] is not None
+               and self.hash_table[index].key != key):
+            index = (index + 1) % self.capacity
+
+        if self.hash_table[index] is None:
+            raise KeyError(key)
+
+        return self.hash_table[index].value
+
+    def __len__(self) -> int:
+        return self.length
+
+    def __delitem__(self, key: Hashable) -> None:
+        for i in range(self.length):
+            if (self.hash_table[i] is not None
+                    and self.hash_table[i].key == key):
+                self.hash_table[i] = None
+
+    def keys(self) -> Any:
+        for i in range(len(self.hash_table)):
+            if self.hash_table[i] is not None:
+                self.dict_keys.append(self.hash_table[i].key)
+        return self.dict_keys
+
+    def values(self) -> Any:
+        for i in range(len(self.hash_table)):
+            if self.hash_table[i] is not None:
+                self.dict_values.append(self.hash_table[i].value)
+        return self.dict_values
+
+    def items(self) -> list:
+        return [(self.hash_table[i].key, self.hash_table[i].value) for i in
+                range(len(self.hash_table)) if self.hash_table[i] is not None]
+
+    def clear(self) -> None:
+        self.hash_table = []
+        self.dict_keys = []
+        self.dict_values = []
+        self.length = 0
+
+    def get(self, key: Hashable) -> Any:
+        return self.__getitem__(key)
+
+    def pop(self, _key: Any) -> Any:
+        for i in range(len(self.hash_table)):
+
+            if (self.hash_table[i] is not None
+                    and self.hash_table[i].key == _key):
+                return self.hash_table[i].value
+
+    def update(self, new_dict: Dictionary) -> Dictionary:
+        for key, value in new_dict.items():
```

Comment:
just add "self[key] = value" here and  remove other lines below. 
it will work the same way because u already defined "__setitem__"
no need to return anything

Reply:
Thanks, it works.

Reply:
this method can be simplified further (in 3 lines) :)
just add "self[key] = value" to line 116, and everything else can be deleted
no need to return anything, just update current dict)

Reply:
Done